miden-tx/src/testing/tx_context.rs

```diff
@@ -0,0 +1,216 @@
+// TRANSACTION CONTEXT
+// ================================================================================================
+
+use alloc::vec::Vec;
+
+use miden_lib::transaction::{ToTransactionKernelInputs, TransactionKernel};
+use miden_objects::{
+    accounts::{
+        account_id::testing::ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_OFF_CHAIN, Account,
+        AccountCode,
+    },
+    assembly::Assembler,
+    notes::{Note, NoteId},
+    testing::{
+        account::MockAccountType,
+        block::MockChain,
+        notes::{mock_notes, AssetPreservationStatus},
+    },
+    transaction::{
+        InputNote, InputNotes, OutputNote, PreparedTransaction, TransactionArgs, TransactionInputs,
+    },
+    FieldElement,
+};
+use vm_processor::{AdviceInputs, ExecutionError, Felt, Process, Word};
+
+use super::{executor::CodeExecutor, utils::create_test_chain, MockHost};
+
+pub struct TransactionContext {
+    mock_chain: MockChain,
+    expected_output_notes: Vec<Note>,
+    tx_args: TransactionArgs,
+    tx_inputs: TransactionInputs,
+}
+
+impl TransactionContext {
+    pub fn execute_code(&self, code: &str) -> Result<Process<MockHost>, ExecutionError> {
+        self.execute_with_inputs(code, AdviceInputs::default())
+    }
+
+    pub fn execute_with_inputs(
+        &self,
+        code: &str,
+        inputs: AdviceInputs,
+    ) -> Result<Process<MockHost>, ExecutionError> {
+        let tx = self.get_prepared_transaction(code);
+        let (stack_inputs, mut advice_inputs) = tx.get_kernel_inputs();
+        advice_inputs.extend(inputs);
+
+        CodeExecutor::new(MockHost::new(tx.account().into(), advice_inputs))
+            .stack_inputs(stack_inputs)
+            .run(code)
+    }
```

Comment:
I'm not sure we need `execute_with_inputs()` in addition to `execute_code()`. It seems like the only reason we have `execute_with_inputs()` is to set an invalid account seed in `test_prologue_create_account_invalid_seed`. If that's the case, a better way would probably be to handle this in the `TransactionContextBuilder`.

Reply:
Removed this